File: enode/src/main/java/com/enode/commanding/CommandResult.java
package com.enode.commanding;

public class CommandResult {
    /**
     * Represents the result status of the command.
     */
    private CommandStatus status;
    /**
     * Represents the unique identifier of the command.
     */
    private String commandId;
    /**
     * Represents the aggregate root id associated with the command.
     */
    private String aggregateRootId;
    /**
     * Represents the command result data.
     */

    private String result;
    /**
     * Represents the command result data type.
     */
    private String resultType;

    public CommandResult() {
    }

    /**
     * Parameterized constructor.
     *
     * @param status
     * @param commandId
     * @param aggregateRootId
     * @param result
     * @param resultType
     */
    public CommandResult(CommandStatus status, String commandId, String aggregateRootId, String result, String resultType) {
        this.status = status;
        this.commandId = commandId;
        this.aggregateRootId = aggregateRootId;
        this.result = result;
        this.resultType = resultType;
    }

    @Override
    public String toString() {
        return String.format("[CommandId=%s,Status=%s,AggregateRootId=%s,Result=%s,ResultType=%s]",
                commandId,
                status,
                aggregateRootId,
                result,
                resultType);
    }

    public CommandStatus getStatus() {
        return status;
    }

    public String getCommandId() {
        return commandId;
    }

    public String getAggregateRootId() {
        return aggregateRootId;
    }

    public String getResult() {
        return result;
    }

    public String getResultType() {
        return resultType;
    }
}

File: enode/src/main/java/org/enodeframework/commanding/CommandResult.kt
package org.enodeframework.commanding

/**
 * @author anruence@gmail.com
 */
class CommandResult {
    /**
     * Represents the result status of the command.
     */
    var status: CommandStatus = CommandStatus.Failed
        private set

    /**
     * Represents the unique identifier of the command.
     */
    var commandId: String = ""
        private set

    /**
     * Represents the aggregate root id associated with the command.
     */
    var aggregateRootId: String = ""
        private set

    /**
     * Represents the command result data.
     */
    var result: String? = ""
        private set

    /**
     * Represents the command result data type.
     */
    var resultType: String = ""
        private set

    constructor()

    /**
     * Parameterized constructor.
     */
    constructor(status: CommandStatus, commandId: String, aggregateRootId: String, result: String?, resultType: String) {
        this.status = status
        this.commandId = commandId
        this.aggregateRootId = aggregateRootId
        this.result = result
        this.resultType = resultType
    }
}